content/operate/kubernetes/security/vault.md

```diff
@@ -0,0 +1,530 @@
+
+1. Create database password in Vault:
+   ```bash
+   vault kv put -namespace=<VAULT_NAMESPACE> \
+     <VAULT_SECRET_ROOT>/redisenterprise-<K8S_NAMESPACE>/redb-<DATABASE_NAME> \
+     password=<DATABASE_PASSWORD>
+   ```
+
+   <br>
+
+2. Create the REDB custom resource:
+   Follow the standard [database creation process]({{< relref "/operate/kubernetes/re-databases" >}}). The REC configuration automatically enables Vault integration for all databases.
+
+   <br>
+
+3. Configure additional secrets (optional):
+   Store additional REDB secrets in the path `redisenterprise-<K8S_NAMESPACE>/`. Secrets must comply with the [REDB secrets schema]({{< relref "/operate/kubernetes/reference/api/redis_enterprise_database_api" >}}).
+
+{{<note>}}
+When using the Redis Enterprise Vault plugin, set `defaultUser: false` and associate users through ACL bindings to the REDB.
+{{</note>}}
+
+For complete field documentation, see the [Redis Enterprise database API reference]({{< relref "/operate/kubernetes/reference/api/redis_enterprise_database_api" >}}).
+
+### Redis Enterprise Remote Cluster secrets
+
+The `secretName` field is supported and should be stored in HashiCorp Vault when the Redis Enterprise cluster uses Vault as a secret source.
+
+### Redis Enterprise Active-Active database secrets
+
+REAADB resources include REDB specifications in the `globalConfigurations` field. All secret names specified in these configurations are supported and should be stored in HashiCorp Vault when the Redis Enterprise cluster uses Vault as a secret source.
+
+## Manage secrets
+
+{{<note>}}
+Complete field documentation is available in the [`RedisEnterpriseCluster` API reference]({{< relref "/operate/kubernetes/reference/api/redis_enterprise_cluster_api" >}}) and [`RedisEnterpriseDatabase` API reference]({{< relref "/operate/kubernetes/reference/api/redis_enterprise_database_api" >}}).
+{{</note>}}
+
+### Redis Enterprise cluster secrets
+
+#### Example REC configuration with all certificates
+
+```yaml
+apiVersion: app.redislabs.com/v1
+kind: RedisEnterpriseCluster
+metadata:
+  name: rec
+  labels:
+    app: redis-enterprise
+spec:
+  nodes: 3
+  licenseSecretName: <VAULT_SECRET_NAME>
+  clusterCredentialSecretName: <VAULT_SECRET_NAME>
+  certificates:
+    apiCertificateSecretName: <VAULT_SECRET_NAME>
+    cmCertificateSecretName: <VAULT_SECRET_NAME>
+    metricsExporterCertificateSecretName: <VAULT_SECRET_NAME>
+    proxyCertificateSecretName: <VAULT_SECRET_NAME>
+    syncerCertificateSecretName: <VAULT_SECRET_NAME>
+    ldapClientCertificateSecretName: <VAULT_SECRET_NAME>
+  # Vault configuration
+  clusterCredentialSecretType: vault
+  clusterCredentialSecretRole: redis-enterprise-rec-<K8S_NAMESPACE>
+  vaultCASecret: vault-ca-cert
+  podAnnotations:
+    vault.hashicorp.com/auth-path: auth/<AUTH_PATH>
+    vault.hashicorp.com/namespace: <VAULT_NAMESPACE>
+```
+
+You can also update certificates using `kubectl patch`:
+
+```bash
```

Comment:
Same with this code block.
<img width="670" height="128" alt="Screenshot 2025-11-14 at 14 19 32" src="https://github.com/user-attachments/assets/29bc9207-749d-43dd-bfd0-248059f4996c" />